client/src/test/java/com/vesoft/nebula/client/graph/net/TestSession.java

```diff
@@ -173,4 +177,62 @@ public void testReconnectWithMultiServices() {
             pool.close();
         }
     }
+
+    @Test
+    public void testExecuteWithParameter() {
+        System.out.println("<==== testExecuteWithParameter ====>");
+        Runtime runtime = Runtime.getRuntime();
+        NebulaPool pool = new NebulaPool();
+        try {
+            // make sure the graphd2_1 without any sessions
+            String cmd = "docker restart nebula-docker-compose_graphd2_1";
+            Process p = runtime.exec(cmd);
+            p.waitFor(10, TimeUnit.SECONDS);
+            ProcessUtil.printProcessStatus(cmd, p);
+
+            NebulaPoolConfig nebulaPoolConfig = new NebulaPoolConfig();
+            nebulaPoolConfig.setMaxConnSize(6);
+            List<HostAddress> addresses = Arrays.asList(
+                    new HostAddress("127.0.0.1", 9669),
+                    new HostAddress("127.0.0.1", 9670),
+                    new HostAddress("127.0.0.1", 9671));
+            TimeUnit.SECONDS.sleep(15);
+            Assert.assertTrue(pool.init(addresses, nebulaPoolConfig));
+            TimeUnit.SECONDS.sleep(15);
+            Session session = pool.getSession("root", "nebula", true);
+            System.out.println("The address of session is " + session.getGraphHost());
+
+            // test ping
+            Assert.assertTrue(session.ping());
+            // prepare parameters
+            Map<byte[], Value> paramMap = new HashMap<byte[], Value>();
+            paramMap.put("p1".getBytes(), Value.iVal(3));
+            paramMap.put("p2".getBytes(), Value.bVal(true));
+            paramMap.put("p3".getBytes(), Value.fVal(3.3));
+            // test executeWithParameter interface
+            ResultSet resp = session.executeWithParameter(
+                    "RETURN $p1+1,$p2,$p3", paramMap);
```

Comment:
From the perspective of user usage, what’s the difference between parameterized execution and non-parameterized execution?

If users use `executeWithParameter` to execute ngql, them also need to concern the `Value` struct, which may increase the use threshold.

Reply:
ACK.

Reply:
Fixed.